tests/conftest.py

```diff
@@ -1,11 +1,34 @@
 # SPDX-License-Identifier: GPL-3.0-or-later
 import pytest
+import flask_migrate
 
-
+from cachito.web import db as _db
 from cachito.web.wsgi import create_app
 
 
 @pytest.fixture(scope='session')
-def client():
+def app():
+    """Return Flask application for the pytest session."""
+    return create_app('cachito.web.config.TestingConfig')
+
+
+@pytest.fixture(scope='session')
+def client(app):
     """Return Flask application client for the pytest session."""
-    return create_app('cachito.web.config.TestingConfig').test_client()
+    return app.test_client()
+
+
+@pytest.fixture()
+def db(app, tmpdir):
+    """Yields a DB with required app tables but with no records."""
+    with app.app_context():
+        flask_migrate.upgrade()
+        connection = _db.engine.connect()
+        # Clear each table to ensure tests are idempotent. The exception is
+        # the package_manager table which contains required data to run app.
+        for table in _db.metadata.tables.keys():
```

Comment:
It might be easier to just delete the SQLite database file before running `flask_migrate.upgrade()`. Also, does `_db.drop_all()` not work when run before `flask_migrate.upgrade()`?

Reply:
`_db.drop_all()` doesn't seem to re-trigger the upgrade process again. I had it deleting the file initially, but thought cleaning up the data would give us more flexibility. I don't have a strong preference either way, which makes me lean towards leaving it as is :)

Reply:
Although... calling out a table here creates a special case which is not ideal. I'll change it.

Reply:
Done!